### PR TITLE
fby3.5: hd: Modify HSC OCP setting

### PR DESCRIPTION
**Pull request by Yi-Shum**
Summary:
- When HSC is ADM1278:
Default or 1ou is RBF, set HSC_OCP_GPIO1_R high.
If 2ou present, set HSC_OCP_GPIO3_R high, others low.
- When HSC is LTC4282:
Default or 1ou is RBF, set ILIM adjust to 0b001.
If 2ou present, set ILIM adjust to 0b011.

Dependency: #464 

Test Plan:
- Build code: Pass

### CODE REVIEW

File: common/shell/shell_platform.h [hunk 1]
```diff
@@ -67,16 +67,18 @@ const char *const sensor_type_name[] = {
...
-				ltc4282) sensor_name_to_num(fan) sensor_name_to_num(tmp431)
-				sensor_name_to_num(pmic) sensor_name_to_num(
-					ina233) sensor_name_to_num(isl69254)
-					sensor_name_to_num(max16550a) sensor_name_to_num(
-						ina230) sensor_name_to_num(xdpe12284c)
-						sensor_name_to_num(raa229621) sensor_name_to_num(
-							nct7718w) sensor_name_to_num(ltc4286)
-							sensor_name_to_num(amd_tsi)
-								sensor_name_to_num(apml_mailbox)
+			sensor_name_to_num(xdpe15284) sensor_name_to_num(ltc4282) sensor_name_to_num(
```
Comment (garnermic) [inline common/shell/shell_platform.h:70]: Please reformat this to make it easier to read.

File: meta-facebook/yv35-hd/src/platform/plat_class.c [hunk 2]
```diff
@@ -296,5 +306,44 @@ void init_platform_config()
...
+	/* Set HSC OCP */
+	uint8_t ilim_adjust = ILIM_ADJUST_DEFAULT;
+	if (_2ou_status.present) {
+		if (hsc_module == HSC_MODULE_ADM1278) {
+			gpio_set(HSC_OCP_GPIO1_R, GPIO_LOW);
+			gpio_set(HSC_OCP_GPIO2_R, GPIO_LOW);
+			gpio_set(HSC_OCP_GPIO3_R, GPIO_HIGH);
+		} else if (hsc_module == HSC_MODULE_LTC4282) {
+			ilim_adjust = ILIM_ADJUST_2OU_EXP;
+		}
```
Comment (garnermic) [inline meta-facebook/yv35-hd/src/platform/plat_class.c:318]: Please add else clause to catch invalid hsc_module values.

File: meta-facebook/yv35-hd/src/platform/plat_class.c [hunk 3]
```diff
@@ -296,5 +306,44 @@ void init_platform_config()
...
+	} else if (_1ou_status.present && (_1ou_status.card_type != TYPE_1OU_RAINBOW_FALLS)) {
+		printf("Unsupported configuration, 1ou card type %d\n", _1ou_status.card_type);
+	}
```
Comment (garnermic) [inline meta-facebook/yv35-hd/src/platform/plat_class.c:321]: Please add else statement or change else-if to just an else statement if it handles all other cases.

**Review by garnermic — CHANGES_REQUESTED** [review]
Few minor changes requested.

File: common/shell/shell_platform.h (hunk 1 above)
Comment (Yi-Shum) [reply common/shell/shell_platform.h:70]: I'll disable clang-format of this array and reformat it.

File: meta-facebook/yv35-hd/src/platform/plat_class.c (hunk 2 above)
Comment (Yi-Shum) [reply meta-facebook/yv35-hd/src/platform/plat_class.c:318]: An else statement will be added to handle all conditions.

File: meta-facebook/yv35-hd/src/platform/plat_class.c (hunk 3 above)
Comment (Yi-Shum) [reply meta-facebook/yv35-hd/src/platform/plat_class.c:321]: An else statement will be added to handle all conditions.

**Comment by Yi-Shum** [comment]
Hi @garnermic ,
I modified it based on the comments.
Please review again, thank you.

**Comment by Yi-Shum** [comment]
Close this PR since HSC OCP setting will be changed.

**Pull request closed without merging**